pkg/build/controller/strategy/custom.go

```diff
@@ -22,12 +23,22 @@ type CustomBuildStrategy struct {
 
 // CreateBuildPod creates the pod to be used for the Custom build
 func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	data, err := bs.Codec.Encode(build)
```

Comment:
was this previously encoding the internal version? that should be fixed regardless of whether strategy.BuildAPIVersion is set.